src/common/mfu_io.c

```diff
@@ -558,6 +1092,35 @@ int mfu_rmdir(const char* dir)
     return rc;
 }
 
+#define NUM_DIRENTS 24
+
+#ifdef DAOS_SUPPORT
+struct dfs_mfu_t {
+    dfs_obj_t *dir;
+    struct dirent ents[NUM_DIRENTS];
+    daos_anchor_t anchor;
+    int num_ents;
+};
+#endif
+
+DIR* daos_opendir(const char* dir, mfu_file_t* mfu_file)
+{
+#ifdef DAOS_SUPPORT
+    struct dfs_mfu_t *dirp = NULL;
+    dirp = calloc(1, sizeof(*dirp));
+    if (dirp == NULL) {
+        return NULL;
+    }
+    int rc = dfs_lookup(mfu_file->dfs, dir, O_RDWR, &dirp->dir, NULL, NULL);
+    if (rc) {
+        MFU_LOG(MFU_LOG_ERR, "dfs_lookup %s failed", dir);
+        errno = ENOENT;
+        return NULL;
```

Comment:
I think we should also free the dirp that was calloc'd in this case

Reply:
@adammoody it is freed in the daos_closedir function

Reply:
Yeah, but in this error case, I don't think closedir would be called, since the opendir returned a failure.

Reply:
@adammoody ah yeah ok I see what you mean

Reply:
@adammoody added the free call